static/js/filter.js

```diff
@@ -99,6 +99,14 @@ function message_matches_search_term(message, operator, operand) {
     case 'sender':
         return people.id_matches_email_operand(message.sender_id, operand);
 
+    case 'huddle-with':
+        var operand_id = people.get_user_id(operand);
+        if (!operand_id) {
+            return false;
+        }
+
+        return (operand_id === message.sender_id);
```

Comment:
I don't think these are the semantics we want.  Let's say I'm Alice, and I've had a huddle conversation with Bob and Cindy recently.  If I say huddle_with:bob@example.com, it's possible that a matching message will actually have been sent by me or Cindy.